src/bin/exrmetrics/main.cpp

```diff
@@ -0,0 +1,203 @@
+
+//
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) Contributors to the OpenEXR Project.
+//
+
+#include "exrmetrics.h"
+
+#include "ImfCompression.h"
+#include "ImfMisc.h"
+
+#include <iostream>
+#include <vector>
+
+#include <math.h>
+#include <stdlib.h>
+#include <string.h>
+
+using std::cerr;
+using std::cout;
+using std::endl;
+using std::ostream;
+using std::vector;
+using namespace Imf;
+
+void
+usageMessage (ostream& stream, const char* program_name, bool verbose = false)
+{
+    stream << "Usage: " << program_name << " [options] infile outfile" << endl;
+
+    if (verbose)
+    {
+        std::string compressionNames;
+        getCompressionNamesString ("/", compressionNames);
+        stream
+            << "Read a part OpenEXR image from infile, write an identical copy to outfile"
+               " reporting time taken to read/write and file sizes.\n"
+               "\n"
+               "Options:\n"
+               "\n"
+               "  -p n          part number to copy (only one part will be written to output file)\n"
```

Comment:
Any reason to not just do all parts? Seems like in the field you'd like to run this on all parts, and observe the effect on the files as a whole, even if the compression stats are specific to individual parts.

Reply:
It's certainly doable, but there are a bunch of issues: since the parts can have different types, and  different compression codecs, generating meaningful output gets messy. The C++ API doesn't give access to the size of individual compressed tiles/scanlines, so the total file size is quoted instead. It would be tricky to quote the compressed size of each part, but some data only makes sense if it's part-specific. A custom stream, or the Core API, might do a better job of that.

Currently, for all types, exrmetrics reads all the data in one go, then writes them all out, timing the total read and write time, since that should give the most reliable statistics. That gets messy to implement if it has to read data all parts first, then write them all out again, using part-specific code for each type.